app/controllers/github_controller.rb

```diff
@@ -213,12 +213,32 @@ def pullapprove_merge_hook
         return
       end
 
+      unless Dir.exist?('SmokeDetector')
+        `git clone https://github.com/Charcoal-SE/SmokeDetector`
+
+        Dir.chdir('SmokeDetector') do
+          `git config user.name metasmoke`
+          `git config user.email #{AppConfig['github']['username']}`
+
+          File.write '.git/info/attributes', <<~END
+            bad_keywords.txt -text merge=union
+            blacklisted_usernames.txt -text merge=union
+            blacklisted_websites.txt -text merge=union
+            watched_keywords.txt -text merge=union
+          END
+        end
+      end
+
       if !Octokit.client.pull_merged?('Charcoal-SE/SmokeDetector', pr_num)
-        Octokit.client.merge_pull_request(
-          'Charcoal-SE/SmokeDetector',
-          pr_num,
-          '--autopull'
-        )
+        Dir.chdir('SmokeDetector') do
+          ref = pr[:head][:ref]
+
+          `git checkout master; git pull origin master`
+          `git fetch origin #{ref}`
+          `git merge origin/#{ref} --no-ff -m 'Merge pull request ##{pr_num} from Charcoal-SE/#{ref} --autopull'`
```

Comment:
String interpolation here makes me nervous. Is it at all possible to parameterize these in some way? If the GH secret gets out, this could be a direct path to full shell access.

Reply:
I'll change it to a `system` call (which also won't capture stdout as a bonus)